### DISCUSSION TITLE
How to drag and drop an item from a view to the diagram

### DISCUSSION THREAD

**Discussion by shinesamjacob** [eclipse-glsp/glsp#555] [General] [answered]
i have a view that has elements from an ecore model, instead of using the palette I wanted to do DnD from the view to the diagram editor.

**Comment by planger** [comment, marked as answer]
There is no generic base implementation for DnD from other views outside of the diagram, because those other views define their custom transfer objects. However, you can of course implement that based on plain browser technologies; we've done that in a customer project.

Roughly speaking, you need to put your custom object into the data transfer from the drag source and react to the drop event on the diagram element and process your transfer object. Most of it is not really GLSP-specific but more related to [HTML drag and drop](https://developer.mozilla.org/en-US/docs/Web/API/HTML_Drag_and_Drop_API). The GLSP specific thing is then just to dispatch a GLSP action on drop of your object.
